lib/keystore.js

```diff
@@ -102,14 +106,47 @@ var KeyStore = function(mnemonic, pwDerivedKey, hdPathString) {
     // for standard Ethereum accounts.
 
     var hdRoot = new Mnemonic(mnemonic).toHDPrivateKey().xprivkey;
-    this.encHdRootPriv = KeyStore._encryptString(hdRoot, pwDerivedKey);
+    this.encHdRootPriv = encryptString(hdRoot, pwDerivedKey);
 
     var hdRootKey = new bitcore.HDPrivateKey(hdRoot);
     var hdPath = hdRootKey.derive(hdPathString).xprivkey;
-    pathKsData.encHdPathPriv = KeyStore._encryptString(hdPath, pwDerivedKey);
+    pathKsData.encHdPathPriv = encryptString(hdPath, pwDerivedKey);
+  }
+}
+
+KeyStore.createVault = function(opts, cb) {
+  var _this = this;
+
+  // Default hdPathString
+  if (!('hdPathString' in opts)) {
+    opts.hdPathString = "m/0'/0'/0'";
   }
+
+  // Default seed phrase if not specified
+  if (!('seedPhrase' in opts)) {
+    opts.seedPhrase = this.generateRandomSeed();
+  }
+
+  if (!('salt' in opts)) {
+    opts.salt = defaultSalt;
```

Comment:
This does not look correct? Shouldn't we be generating a random salt if the user doesn't pass it in? Now it uses the default salt again.